### DISCUSSION TITLE
`justfile` with multiline strings incorrectly highlighting code

### DISCUSSION THREAD

**Discussion by macintacos** [github-linguist/linguist#6401] [Classification]
Take this example:

```just
test := """
    This is a
    multiline string
"""

recipe:
    echo oops

another:
    echo not a string, but looks like one
```

Notice how everything between the `"""` on line 4 onwards is considered to be a multiline string, when it shouldn't be.

This seems like a clear bug to me - [the above is valid according to the `just` README](https://github.com/casey/just#strings), so GitHub should be highlighting only the first few lines as a string (set to the variable `test`), with the others interpreted as recipes.

This also happens when you have a multiline string between single quotes (which is also valid):

```just
single := '
hello
'

double := "
goodbye
"
```

**Comment by lildude** [comment]
Syntax highlighting problems should be logged with the upstream grammar and not here as per the [docs](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file). If this has already been fixed in the upstream grammar, GitHub will get the fix when the next [release is made](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes).

**Reply by macintacos** [reply]
Sorry for the delay - thanks. It looks like [the repo that that list in the docs points to is actually archived](https://github.com/github-linguist/linguist/blob/master/vendor/README.md#:~:text=Just%3A%20skellock/vscode%2Djust). I guess I'll reach out to the `just` maintainers and ask them to resolve the issue and point to a better repository for this.

**Reply by tgross35** [reply]
Is switching to tree-sitter possible? I am working on getting https://github.com/IndianBoy42/tree-sitter-just updated. It's not perfect but it is _significantly_ better than the currently used archived option, https://github.com/skellock/vscode-just

**Reply by lildude** [reply]
> Is switching to tree-sitter possible?

No, not though linguist. See https://github.com/github-linguist/linguist/discussions/6073

**Reply by tgross35** [reply]
Thanks. If opening discussion at https://github.com/orgs/community/discussions/ is still the correct step, I will do that in a few days.

**Reply by tgross35** [reply]
Created at https://github.com/orgs/community/discussions/116733
